tracing-appender/src/rolling.rs

```diff
@@ -82,8 +86,10 @@ use time::{format_description, Duration, OffsetDateTime, Time};
 /// [make_writer] tracing_subscriber::fmt::writer::MakeWriter
 #[derive(Debug)]
 pub struct RollingFileAppender {
-    state: Inner,
-    writer: RwLock<File>,
+    pub(crate) state: Inner,
+    pub(crate) writer: RwLock<WriterChannel>,
+    #[cfg(features = "compression")]
+    pub(crate) compression: Option<CompressionConfig>,
```

Comment:
if we move the `builder` module into a submodule of this module, as I suggested, I think we can remove the `pub(crate)` from these fields:
```suggestion
    state: Inner,
    writer: RwLock<WriterChannel>,
    #[cfg(features = "compression")]
    compression: Option<CompressionConfig>,
```